certbot/log.py

```diff
@@ -86,40 +87,40 @@ def post_arg_parse_setup(config):
     :param certbot.interface.IConfig config: Configuration object
 
     """
-    file_handler, file_path = setup_log_file_handler(
-        config, 'letsencrypt.log', FILE_FMT)
-    logs_dir = os.path.dirname(file_path)
+    with setup_log_file_handler(config, 'letsencrypt.log', FILE_FMT) as (file_handler, file_path):
```

Comment:
I don't think this works. When we return from this function, won't the lock on the logs directory be released?

Reply:
Indeed. This method also needs to become a context manager.